src/renderer/components/delete-cluster-dialog/delete-cluster-dialog.tsx

```diff
@@ -0,0 +1,195 @@
+
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this, {
+      showContextSwitch: observable,
+      newCurrentContext: observable
+    });
+  }
+
+  static open({ config, cluster }: Partial<DialogState>) {
+    dialogState.isOpen = true;
+    dialogState.config = config;
+    dialogState.cluster = cluster;
+  }
+
+  static close() {
+    dialogState.isOpen = false;
+  }
+
+  @boundMethod
+  onOpen() {
+    this.showContextSwitch = false;
+    this.newCurrentContext = "";
+  }
+
+  onClose() {
+    DeleteClusterDialog.close();
+  }
+
+  removeContext() {
+    dialogState.config.contexts = dialogState.config.contexts.filter(item =>
+      item.name !== dialogState.cluster.contextName
+    );
+  }
+
+  changeCurrentContext() {
+    if (this.newCurrentContext) {
+      dialogState.config.currentContext = this.newCurrentContext;
+    }
+  }
+
+  @boundMethod
+  async onDelete() {
+    const { cluster, config } = dialogState;
+
+    await requestMain(clusterSetDeletingHandler, cluster.id);
+    this.removeContext();
+    this.changeCurrentContext();
+
+    try {
+      await saveKubeconfig(config, cluster.kubeConfigPath);
+      HotbarStore.getInstance().removeAllHotbarItems(cluster.id);
+      await requestMain(clusterDeleteHandler, cluster.id);
+    } catch(error) {
+      Notifications.error(`Cannot remove cluster, failed to process config file. ${error}`);
+      await requestMain(clusterClearDeletingHandler, cluster.id);
+    }
+
+    this.onClose();
+  }
+
+  renderCurrentContextSwitch(contexts: Context[]) {
+    if (!this.showContextSwitch) return null;
+
+    const options = [
+      ...contexts.map(context => ({
+        label: context.name,
+        value: context.name,
+      })),
+    ];
+
+    return (
+      <div className="mt-4">
+        <p className="mb-4 font-semibold">Choose new current-context</p>
+        <Select
+          options={options}
+          onChange={({ value }) => this.newCurrentContext = value}
+          themeName="light"
+          className="ml-[1px] mr-[1px]"
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const { cluster, config, isOpen } = dialogState;
+
+    if (!cluster || !config) return null;
+
+    const isCurrentContext = config.currentContext == cluster.contextName;
+    const contexts = config.contexts.filter(context => context.name !== cluster.contextName);
+
+    const warning = isCurrentContext ? (
+      <>
+        <p data-testid="context-warning">The <b>current-context</b> field from the kubeconfig file indicates a minikube context
+        that will cease to exist after the change. This will affect the operation of kubectl.{" "}
+        {contexts.length > 0 && (
+          <b
+            className="cursor-pointer underline"
+            onClick={() => this.showContextSwitch = !this.showContextSwitch}
+          >Replace current context</b>
+        )}
+        </p>
+      </>
+    ) : (
+      <p>The contents of your kubeconfig file will be changed!</p>
```

Comment:
This may be nit-picky but when there are no other contexts we could give the user the option to "Delete current context" (similar to "Replace current context") . This assumes that the user knows this refers to `current-context` in the kubeconfig. Or "Delete current-context" 

Though either way this leaves the kubeconfig with an empty contexts field, and cripples `kubectl`